app/src/test/java/org/simple/clinic/patient/PatientRepositoryTest.kt

```diff
@@ -31,7 +34,101 @@ class PatientRepositoryTest {
   fun setUp() {
     database = mock()
+    mockPatientSearchResultDao = mock()
     repository = PatientRepository(database, dobValidator)
+
+  }
+
+  @Test
+  @Parameters(value = [
+    "Name, Name, Name",
+    "Name Surname, Name Surname, NameSurname",
+    "Name Surname, Name   Surname , NameSurname",
+    "Old Name, Name-Surname, NameSurname",
+    "Name, Name Middle-Surname, NameMiddleSurname"
+  ])
+  fun `when merging patients with server records, update the searchable name of the patient to the full name stripped of all spaces and punctuation`(
+      localFullName: String,
+      remoteFullName: String,
+      expectedSearchableName: String
+  ) {
+    whenever(database.patientDao()).thenReturn(mockPatientDao)
+    whenever(database.addressDao()).thenReturn(mockPatientAddressDao)
+
+    val patientUUID = UUID.randomUUID()
+    val addressUUID = UUID.randomUUID()
+
+    val localPatientCopy = PatientMocker.patient(uuid = patientUUID, fullName = localFullName, addressUuid = addressUUID, syncStatus = SyncStatus.DONE)
+    whenever(mockPatientDao.getOne(patientUUID)).thenReturn(localPatientCopy)
+
+    val serverAddress = PatientMocker.address(addressUUID).toPayload()
+    val serverPatientWithoutPhone = PatientPayload(
+        uuid = patientUUID,
+        fullName = remoteFullName,
+        gender = mock(),
+        dateOfBirth = mock(),
+        age = 0,
+        ageUpdatedAt = mock(),
+        status = mock(),
+        createdAt = mock(),
+        updatedAt = mock(),
+        address = serverAddress,
+        phoneNumbers = null)
+
+    repository.mergeWithLocalData(listOf(serverPatientWithoutPhone)).blockingAwait()
+
+    verify(mockPatientDao).save(argThat<List<Patient>> { first().searchableName == expectedSearchableName })
+  }
+
+  @Test
+  @Parameters(value = [
+    "Name, Name",
+    "Name   Surname, NameSurname",
+    "Name Middle Surname, NameMiddleSurname",
+    "Name \tSurname, NameSurname"
+  ])
+  fun `when searching for patients without age bound, strip the search query of any whitespace or punctuation`(
+      query: String,
+      expectedSearchQuery: String
+  ) {
+    whenever(mockPatientSearchResultDao.search(any())).thenReturn(Flowable.just(emptyList()))
+    whenever(database.patientSearchDao()).thenReturn(mockPatientSearchResultDao)
+    whenever(database.addressDao()).thenReturn(mockPatientAddressDao)
+
+    repository.searchPatientsAndPhoneNumbers(query).blockingFirst()
+
+    verify(mockPatientSearchResultDao).search(eq(expectedSearchQuery))
+  }
+
+  @Test
+  fun `when searching for patients with age bound, strip the search query of any whitespace or punctuation`() {
+
+    whenever(mockPatientSearchResultDao.search(any(), any(), any())).thenReturn(Flowable.just(emptyList()))
+    whenever(database.patientSearchDao()).thenReturn(mockPatientSearchResultDao)
+    whenever(database.addressDao()).thenReturn(mockPatientAddressDao)
+
+    repository.searchPatientsAndPhoneNumbers("Name   Surname", 40).blockingFirst()
```

Comment:
Tell me if I'm wrong, but can't we move these tests to `PatientRepositoryAndroidTest`? We will then be able to assert on the data directly.

Reply:
We can move these tests, but there are a couple of other test cases related to searching for phone numbers that ensure the digits are not stripped out when we do a search query. Those are currently not handled in `PatientRepositoryAndroidTest`. If we remove them here, we should add more tests to `PatientRepositoryAndroidTest` to verify phone number search.

Reply:
To add to this, the `PatientRepositoryAndroidTest` test cases do not exercise all code paths in the search patients methods.

Reply:
I haven't moved any of the tests so far, just waiting for your opinion on whether we should move them or not.

Reply:
I'd prefer testing on data. But I think it's okay for now. :)